plutus-ir/src/Language/PlutusIR/Compiler/Term.hs

```diff
@@ -18,7 +18,7 @@ import qualified Language.PlutusCore.MkPlc            as PLC
 
 import           Data.List
 
--- | Compile a 'Term' into a PLC Term.
+-- | Compile a 'Term' into a PLC Term. Note: the result does *not* have globally unique names.
```

Comment:
Just apply `rename` in the end?

Reply:
It's also annoying because `rename` produces different errors (of course).

Reply:
`rename` does not produce any errors, because it leaves free variables untouched, so there is no room for failing.

> And you lose the property anyway the moment you combine it with other things in `Quote`.

Sorry, I don't understand. `f <$> getThing1 <*> getThing2` passes `thing1` and `thing2` to `f` and both of them must not contain any duplicate names and the intersection of sets of their bound variables must be empty, right?

Reply:
> rename does not produce any errors, because it leaves free variables untouched, so there is no room for failing.

Ah, I'd missed that that changed. That does make things easier!

> Sorry, I don't understand.

You're right. I'll just replace the exported `compileTerm` with one that also does renaming.